src/main.rs

```diff
@@ -9,33 +9,25 @@ mod gui;
 
 use ajour_core::fs::CONFIG_DIR;
 use ajour_core::utility::{remove_file, rename};
+use color_eyre::eyre::{Report, Result, WrapErr};
 
 #[cfg(target_os = "linux")]
-use anyhow::Context;
 use std::env;
-use std::path::PathBuf;
+use std::path::{Path, PathBuf};
 
 pub const VERSION: &str = env!("CARGO_PKG_VERSION");
 
-pub type Result<T, E = anyhow::Error> = std::result::Result<T, E>;
-
-pub fn main() {
+pub fn main() -> color_eyre::eyre::Result<()> {
+    color_eyre::install()?;
     let opts_result = cli::get_opts();
 
-    #[cfg(debug_assertions)]
-    let is_debug = true;
-    #[cfg(not(debug_assertions))]
-    let is_debug = false;
+    let is_debug = cfg!(debug_assertions);
 
     // If this is a clap error, we map to None since we are going to exit and display
     // an error message anyway and this value won't matter. If it's not an error,
     // the underlying `command` will drive this variable. If a `command` is passed
     // on the command line, Ajour functions as a CLI instead of launching the GUI.
-    let is_cli = opts_result
-        .as_ref()
-        .map(|o| &o.command)
-        .unwrap_or(&None)
-        .is_some();
+    let is_cli = opts_result.as_ref().map(|o| &o.command).is_ok();
```

Comment:
We're not trying to evaluate if the CLI input was properly deserialized (the Result), we are trying to evaluate if a Command was passed on the CLI (the underlying Option). This is needed inside the `validate_opts_or_exit` to properly implement a workaround for windows to attach to the parent console when built / run in release mode.

We need to evaluate this before evaluating if `opts_result` is an error so we can implement the workaround for windows prior to printing the error message to the command line so it'll show up.

Reply:
Fixed in: f97dc59011407aa855192ac0ace0f61a8ada0521